### PR TITLE
[AM][OPS-9414] Update 'Change' link and add and display income figures

### PR DESCRIPTION
**Pull request by almorcrette**
ticket: https://jira.tools.tax.service.gov.uk/browse/OPS-9414

### CODE REVIEW

File: app/views/partials/income_list.scala.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+@import ssttpaffordability.model.IncomeCategory
+
+@(income: Seq[IncomeCategory])(implicit messages: play.api.i18n.Messages)
+
+<div class="govuk-!-padding-left-6 govuk-!-margin-bottom-9">
+    <dl class="govuk-summary-list">
+
+        @{income.map( category => if (category.amount > 0)
+            <div class="govuk-summary-list__row">
+                <dt class="govuk-summary-list__key govuk-!-font-weight-regular govuk-!-width-one-half">{messages("ssttp.affordability.your-monthly-income.form."+{s"${category.name}"})}</dt>
```
Comment (andy1138) [inline app/views/partials/income_list.scala.html]: `s"${category.name}"` is just `category.name` or `category.name.toString` 

Also, creating message keys by concatenating strings is known to go wrong over time, ie add a new income type without knowing you have to have something in messages.cy etc.  And Intellij cannot analyze the code and say that a value in. messages.cy is being used.
Worth considering something like `category.messageKey` and having the complete key stored with the item.  
( only a suggestion, nothing wrong with the code)

File: test/pagespecs/pages/AddIncomeSpendingPage.scala [hunk 2]
```diff
@@ -19,36 +19,79 @@ package pagespecs.pages
...
   }
 
-  def clickContinue(): Unit = {
-    clickOnContinue()
+  def assertAddIncomeLinkIsDisplayed(implicit lang: Language = Languages.English): Unit = probing {
+    assertContentMatchesExpectedLines(Seq(Expected.LinkText.AddIncome()))
   }
 
-  override def assertPageIsDisplayed(implicit lang: Language): Unit = probing {
+  def assertIncomeTableDisplayed(categoriesFilled: IncomeCategory*)(implicit lang: Language): Unit = {
```
Comment (andy1138) [inline test/pagespecs/pages/AddIncomeSpendingPage.scala:46]: `IncomeCategory*` somebody has been reading the manual :-)

File: app/views/partials/income_list.scala.html (hunk 1 above)
Comment (almorcrette) [reply app/views/partials/income_list.scala.html]: Yep, that makes sense, let me change that

File: app/views/partials/income_list.scala.html [hunk 3]
```diff
@@ -23,7 +23,7 @@
 
         @{income.map( category => if (category.amount > 0)
             <div class="govuk-summary-list__row">
-                <dt class="govuk-summary-list__key govuk-!-font-weight-regular govuk-!-width-one-half">{messages("ssttp.affordability.your-monthly-income.form."+{s"${category.name}"})}</dt>
+                <dt class="govuk-summary-list__key govuk-!-font-weight-regular govuk-!-width-one-half">{messages(s"${category.messageKey}")}</dt>
```
Comment (andy1138) [inline app/views/partials/income_list.scala.html]: `s"${category.messageKey}")`  is just `category.messageKey`

**Review by andy1138 — APPROVED** [review]

**Pull request merged**